ext/stackprof/stackprof.c

```diff
@@ -742,6 +745,9 @@ Init_stackprof(void)
     _stackprof.empty_string = rb_str_new_cstr("");
     rb_global_variable(&_stackprof.empty_string);
 
+    _stackprof.empty_hash = rb_hash_new();
+    rb_global_variable(&_stackprof.empty_hash);
+
```

Comment:
Also, I don't think it is worth adding a test for this, but if anyone feels strongly that we should do one, I can do that. It'll be similar to the following:

```ruby
def test_gc_does_not_segfault_with_out_path_defined
  StackProf.run(interval: 100, out: '/tmp/stackprof-cpu.dump') do
    GC.start
  end
end
```